src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
```

Comment:
I agree, it *should*, but I weighed this against separation of concerns. If I put this in ExecuteRequestParamsBase, the ExecuteDocumentSelectionParams class would need access to the WorkspaceService. This isn't ideal, since really only the *Service classes should interact with the other services. On the other hand, if we pass in the WorkspaceService to the GetSqlText method on the ExecuteDocumentSelectionParams, we'd also have to pass it into the ExecuteStringParams implementation, which isn't useful.

In this case, my opinion is that we should do this kinda annoying cast in order to preserve the contracts from not needing to know about the services that utilize them.